### ISSUE TITLE
Wrong pseudo random number generation in JAX Extra Trees implementation

### ISSUE THREAD

**Issue by andruekonst** [NTAILab/bosk#71]
Decision trees use random numbers which are not populated from some input random seed, which **should be different** for trees in an ensemble:
https://github.com/NTAILab/bosk/blob/a3dc02242381a9eb8cf2d038e34c8fb5d7269250/bosk/block/zoo/models/classification/_jax_util.py#L88

At ensemble level, the data split does not depend on a user specified random seed:
https://github.com/NTAILab/bosk/blob/a3dc02242381a9eb8cf2d038e34c8fb5d7269250/bosk/block/zoo/models/classification/jax.py#L115
